bootstrapvz/common/tasks/grub.py

```diff
@@ -25,14 +25,52 @@ class ConfigureGrub(Task):
 	def run(cls, info):
 		from bootstrapvz.common.tools import sed_i
 		grub_def = os.path.join(info.root, 'etc/default/grub')
+
+		# disable the graphical grub menu
 		sed_i(grub_def, '^#GRUB_TERMINAL=console', 'GRUB_TERMINAL=console')
-		sed_i(grub_def, '^GRUB_CMDLINE_LINUX_DEFAULT="quiet"',
-		                'GRUB_CMDLINE_LINUX_DEFAULT="console=hvc0"')
-		sed_i(grub_def, '^GRUB_TIMEOUT=[0-9]+', 'GRUB_TIMEOUT=0\n'
-		                                        'GRUB_HIDDEN_TIMEOUT=0\n'
-		                                        'GRUB_HIDDEN_TIMEOUT_QUIET=true')
+
+		# optional grub settings
+		grub_settings = {}
+		if 'grub' in info.manifest.system:
+			grub_settings = info.manifest.system['grub']
+
+		# enable grub menu
+		if grub_settings.get('enable_timeout') > 0:
+			sed_i(
+				grub_def,
+				'^GRUB_TIMEOUT=[0-9]+',
+				'GRUB_TIMEOUT=' + str(grub_settings.get('enable_timeout'))
+			)
+		else:
+			sed_i(
+				grub_def,
+				'^GRUB_TIMEOUT=[0-9]+',
+				'GRUB_TIMEOUT=0\n'
+				'GRUB_HIDDEN_TIMEOUT=0\n'
+				'GRUB_HIDDEN_TIMEOUT_QUIET=true'
+			)
+
+		# disable persistent network interface names
+		if grub_settings.get('disable_pnin') is True:
+			sed_i(
+				grub_def,
+				'^GRUB_CMDLINE_LINUX=""',
+				'GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0"'
+			)
+
+		# quiet the console
+		sed_i(
+			grub_def,
+			'^GRUB_CMDLINE_LINUX_DEFAULT="quiet"',
+			'GRUB_CMDLINE_LINUX_DEFAULT="console=hvc0 console=ttyS0"'
```

Comment:
I saw it mentioned here:
https://github.com/andsens/bootstrap-vz/pull/256
I had issues locally, when manually starting out a Virtualbox instance, outside of Vagrant, the console would appear to hang, when it was actually being rerouted. This prevented me from diagnosing issues. This changes fixes that.